src/runtime.js

```diff
@@ -7,56 +8,101 @@ var __hasOwnProp = Object.prototype.hasOwnProperty;
 var __getOwnPropNames = Object.getOwnPropertyNames;
 var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
 
-// We're disabling Object.freeze because it breaks CJS => ESM and can cause
-// issues with Suspense and other things that expect the CJS module namespace
-// to be mutable when the ESM module namespace is NOT mutable
-// var __objectFreezePolyfill = new WeakSet();
+export var __markAsModule = target => __defProp(target, "__esModule", { value: true, configurable: true });
 
-// globalThis.Object.freeze = function freeze(obj) {
-//   __objectFreezePolyfill.add(obj);
-//   return obj;
-// };
+// This is used to implement "export * from" statements. It copies properties
+// from the imported module to the current module's ESM export object. If the
+// current module is an entry point and the target format is CommonJS, we
+// also copy the properties to "module.exports" in addition to our module's
+// internal ESM export object.
+export var __reExport = (target, mod, secondTarget) => {
+  for (let key of __getOwnPropNames(mod))
+    if (!__hasOwnProp.call(target, key) && key !== "default")
+      __defProp(target, key, {
+        get: () => mod[key],
+        enumerable: true,
+      });
 
-// globalThis.Object.isFrozen = function isFrozen(obj) {
-//   return __objectFreezePolyfill.has(obj);
-// };
+  if (secondTarget) {
+    for (let key of __getOwnPropNames(mod))
+      if (!__hasOwnProp.call(secondTarget, key) && key !== "default")
+        __defProp(secondTarget, key, {
+          get: () => mod[key],
+          enumerable: true,
+        });
 
-export var __markAsModule = target => __defProp(target, "__esModule", { value: true, configurable: true });
+    return secondTarget;
+  }
+};
+
+// Converts the module from CommonJS to ESM. When in node mode (i.e. in an
+// ".mjs" file, package.json has "type: module", or the "__esModule" export
+// in the CommonJS file is falsy or missing), the "default" property is
+// overridden to point to the original CommonJS exports object instead.
+export var __toESM = (mod, isNodeMode, target) => {
+  target = mod != null ? __create(__getProtoOf(mod)) : {};
+  const to =
+    isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target;
+
+  // If the importer is in node compatibility mode or this is not an ESM
+  // file that has been converted to a CommonJS file using a Babel-
+  // compatible transform (i.e. "__esModule" has not been set), then set
+  // "default" to the CommonJS "module.exports" for node compatibility.
+  for (let key of __getOwnPropNames(mod))
+    if (!__hasOwnProp.call(to, key))
+      __defProp(to, key, {
+        get: () => mod[key],
+        enumerable: true,
+      });
+
+  return to;
+};
+
+/* @__PURE__ */
+var moduleCache = new /* @__PURE__ */ WeakMap();
```

Comment:
The pure annotations above are in the wrong place. They must directly precede a function call or `new`.

```sh
$ echo '/*@__PURE__*/ var obj = new /*@__PURE__*/ FooBar();' | uglify-js --toplevel -bmc
new FooBar();

$ echo '/*@__PURE__*/ var obj = new /*@__PURE__*/ FooBar();' | rollup --silent
new  FooBar();

$ echo '/*@__PURE__*/ var obj = new /*@__PURE__*/ FooBar();' | esbuild --bundle --minify
(()=>{var o=new FooBar;})();
```

```sh
$ echo 'var obj = /*@__PURE__*/ new FooBar();' | uglify-js --toplevel -bmc

$ echo 'var obj = /*@__PURE__*/ new FooBar();' | rollup --silent

$ echo 'var obj = /*@__PURE__*/ new FooBar();' | esbuild --bundle --minify
(()=>{})();
```

Side question: Any way `bun` could accept javascript input from stdin?

/cc @alexlamsl

Reply:
> Any way `bun` could accept javascript input from stdin?

I'd certainly vote for that &minus; makes hand-testing of cross-platform compatibility a _lot_ easier :nerd_face:

Will look into that during my next spare cycles...